src/Microsoft.IdentityModel.Tokens.Saml/Saml2/Saml2SecurityTokenHandler.cs

```diff
@@ -420,7 +420,7 @@ private Saml2SecurityToken ValidateSignature(Saml2SecurityToken samlToken, strin
                 {
                     try
                     {
-                        samlToken.Assertion.Signature.Verify(key, validationParameters.CryptoProviderFactory ?? key.CryptoProviderFactory);
+                        samlToken.Assertion.Signature.Verify(key, validationParameters.CryptoProviderFactory ?? key.CryptoProviderFactory, samlToken, validationParameters);
```

Comment:
We could just verify that the key and algorithm are valid before calling Signature.Verify.
The reasoning is that an XML Signature may or not be associated with a token. 
TokenValidationParameters may not the right object conceptually.

thoughts?

Reply:
We can do that, and it would simplify the changes a bit. I think that would also address the issue you brought up about the conceptual mapping?

Reply:
@keegan-caruso super....  ping me when the changes are in....

Reply:
@brentschmaltz changes are in and build is passing